### PR TITLE
OEL-1551: Add documents field for person content type.

### PR DESCRIPTION
**Pull request by donquixote**

### CODE REVIEW

**Review by escuriola — DISMISSED** [review]

File: modules/oe_starter_content_person/config/install/field.storage.node.oe_sc_person_documents.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+id: node.oe_sc_person_documents
+field_name: oe_sc_person_documents
+entity_type: node
+type: entity_reference
+settings:
+  target_type: oe_document_reference
+module: core
+locked: false
+cardinality: -1
+translatable: true
```
Comment (drishu) [inline modules/oe_starter_content_person/config/install/field.storage.node.oe_sc_person_documents.yml:16]: this should be not translatable, like here https://github.com/openeuropa/oe_content/blob/3.x/modules/oe_content_person/config/install/field.storage.node.oe_person_documents.yml

File: composer.json [hunk 2]
```diff
@@ -50,6 +51,11 @@
     "extra": {
         "composer-exit-on-patch-failure": true,
         "enable-patching": true,
+        "patches": {
+            "openeuropa/oe_content": {
```
Comment (drishu) [inline composer.json]: we are using oe_content v2 but patching against v3, not good, this breaks at install for me even if the patch applies cleanly.
We should use the dev branch and epic once merged; 3.x once epic merged; and stable once released.

File: modules/oe_starter_content_person/config/install/core.entity_form_display.node.oe_sc_person.default.yml [hunk 3]
```diff
@@ -47,6 +49,24 @@ content:
...
+      override_labels: false
+      label_singular: ''
+      label_plural: ''
+      allow_new: true
+      allow_existing: false
+      match_operator: CONTAINS
+      allow_duplicate: false
+      collapsible: false
+      collapsed: false
+      revision: false
```
Comment (drishu) [inline modules/oe_starter_content_person/config/install/core.entity_form_display.node.oe_sc_person.default.yml]: I see in oe_content its revision true and removed_reference keep, but.. 
Why keep these entities if you can't select any existing ones ?

File: tests/src/Traits/ExampleMediaTrait.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
+<?php
```
Comment (drishu) [inline tests/src/Traits/ExampleMediaTrait.php]: I don't like some of this stuff that much, as I suspected in 2020 there is and will be a need for these helper methods
https://github.com/openeuropa/oe_media/pull/124/files
I think this is rather the way to go, but I don't know if we can get that merged in or not

File: modules/oe_starter_content_person/config/install/core.entity_form_display.node.oe_sc_person.default.yml (hunk 3 above)
Comment (drishu) [reply modules/oe_starter_content_person/config/install/core.entity_form_display.node.oe_sc_person.default.yml]: @brummbar do you have any insight on this ?

File: tests/src/Traits/ExampleMediaTrait.php (hunk 4 above)
Comment (drishu) [reply tests/src/Traits/ExampleMediaTrait.php]: @brummbar what do you think ?

**Review by drishu — CHANGES_REQUESTED** [review]

File: modules/oe_starter_content_person/config/install/core.entity_form_display.node.oe_sc_person.default.yml (hunk 3 above)
Comment (brummbar) [reply modules/oe_starter_content_person/config/install/core.entity_form_display.node.oe_sc_person.default.yml]: It needs to be "keep" to work with revisions. If we delete the referenced entity, restoring old revisions of documents won't work correctly.

File: modules/oe_starter_content_person/config/install/field.storage.node.oe_sc_person_documents.yml (hunk 1 above)
Comment (brummbar) [reply modules/oe_starter_content_person/config/install/field.storage.node.oe_sc_person_documents.yml:16]: oe_content has both translatable (6) and untranslatable (33) storages for entity references.
Thing is, normally entity references are not meant to be translated but this varies case by case.
In DGCOMM content types we can assume there is no need of having different values for translations of entity references, but we cannot make the same assumption for other projects.
So we should make these storages always translatable, as it gives the maximum flexibility.
We then can mark the instance as not translatable. If the users of the module need it translated, it's very easy to change the flag, while it's much more complex to fix the storage.
So I'd say, let's keep the storages translatable.

File: tests/src/Traits/ExampleMediaTrait.php (hunk 4 above)
Comment (brummbar) [reply tests/src/Traits/ExampleMediaTrait.php]: I don't like the idea of putting a dependency on test code coming from another module. I don't even much like the trait in se, we use it in one class only, we could have created simple protected methods.
Let's avoid to have to deal with oe_media too, let's put back the code inside our test class. We will move it when we need it.

**Review by drishu — DISMISSED** [review]

File: composer.json [hunk 5]
```diff
@@ -8,6 +8,7 @@
     "require": {
         "php": ">=7.4",
         "drupal/address": "^1.10",
+        "drupal/composite_reference": "^2.1",
```
Comment (brummbar) [inline composer.json]: This belongs to the require-dev section, as it's required only in some submodules. We should then add in the README of the single submodule which parts are required.
See https://github.com/openeuropa/oe_content/tree/2.x/modules/oe_content_consultation#installation

File: modules/oe_starter_content_person/config/install/field.field.node.oe_sc_person.oe_sc_person_documents.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    - field.storage.node.oe_sc_person_documents
+    - node.type.oe_sc_person
+    - oe_content_sub_entity_document_reference.oe_document_reference_type.oe_document
+    - oe_content_sub_entity_document_reference.oe_document_reference_type.oe_document_group
+  module:
+    - composite_reference
+third_party_settings:
+  composite_reference:
+    composite: true
+    composite_revisions: null
```
Comment (brummbar) [inline modules/oe_starter_content_person/config/install/field.field.node.oe_sc_person.oe_sc_person_documents.yml]: Let's set this also to true.

**Review by brummbar — CHANGES_REQUESTED** [review]

**Review by drishu — DISMISSED** [review]

**Review by brummbar — DISMISSED** [review]

**Review by drishu — DISMISSED** [review]

**Review by brummbar — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**